#15082 API design: Unify API endpoints for downloading MDM files & getting metadata about MDM files (#15267)
fleetdm/fleet: docs/REST API/rest-api.md

`Status: 200`

### Get or download a bootstrap package

_Available in Fleet Premium_

> `GET /mdm/bootstrap/{team_id}`

Author: ```suggestion
`GET /mdm/bootstrap/:team_id`
```

#### Parameters

| Name       | Type    | In    | Description                                      |
| -----------| ------  | ----- | ------------------------------------------------ |
| team_id    | integer | path  | **Required.** The desired bootstrap package's team's ID. |

> | alt        | string  | query | If specified and set to "media", downloads the bootstrap package's contents. |

Reviewer: ```suggestion
| alt        | string  | query | If specified and set to "media", downloads the bootstrap package. |
```
Can we simplify this, do we need "contents"?

> | for_update | boolean | query | If set to `true`, the authorization will be for a `write` action instead of a `read`. Useful for the write-only `gitops` role when requesting the bootstrap metadata to check if the package needs to be replaced. _(Only relevant if `alt` is not specified.)_ |

Author: @noahtalerman wasn't sure if this parameter was still needed so kept it for now. Let me know if I should remove it.

Reviewer 2: @roperzh is this parameter still needed?

> #### Example (get a bootstrap package)

Reviewer: ```suggestion
#### Example (get a bootstrap package metadata)
```

  "name": "bootstrap-package.pkg",
  "team_id": 0,
  "sha256": "6bebb4433322fd52837de9e4787de534b4089ac645b0692dfb74d000438da4a3",
  "created_at": "2023-04-20T13:02:05Z"
}
```

In the response above:
- `sha256` is the SHA256 digest of the bytes of the bootstrap package file.

> #### Example (download bootstrap package's contents)

Reviewer: ```suggestion
#### Example (download a bootstrap package)
```

Author: Was originally following the pattern from the [get script examples](https://fleetdm.com/docs/rest-api/rest-api#get-or-download-a-script) to keep the structure consistent, but I agree it makes more sense this way— I'll update those headings as well so they all match

`Status: 200`

### Get or download an EULA file

_Available in Fleet Premium_

> `GET /mdm/eula/{team_id}`

Author: ```suggestion
`GET /mdm/eula/:team_id`
```

#### Parameters

| Name       | Type    | In    | Description                                      |
| -----------| ------  | ----- | ------------------------------------------------ |
| team_id    | integer | path  | **Required.** The desired EULA file's team's ID. |

> | alt        | string  | query | If specified and set to "media", downloads the EULA file's contents. |

Reviewer: ```suggestion
| alt        | string  | query | If specified and set to "media", downloads the EULA file. |
```
Same as above. I would cut "...'s contents", since it's easier to read without.

> #### Example (get EULA file)

Reviewer: ```suggestion
#### Example (get EULA file metadata)
```

`GET /mdm/eula/123`

##### Example response

`Status: 200`

```json
{
  "name": "eula.pdf",
  "created_at": "2023-04-20T13:02:05Z"
}
```

> #### Example (download EULA file's contents)

Reviewer: ```suggestion
#### Example (download EULA file)
```
